src/Planner/PlannerAggregation.cpp

```diff
@@ -183,6 +185,87 @@ void resolveGroupingFunctions(QueryTreeNodePtr & query_node,
     resolveGroupingFunctions(query_node, group_by_kind, aggregation_keys, grouping_sets_parameters_list, planner_context);
 }
 
+}
+
+void resolveGroupingFunctions(QueryTreeNodePtr & query_node, const PlannerContext & planner_context)
```

Comment:
Why do we resolve `GROUPING` function in the planner, not in QueryAnalysisPass? I'd recommend moving this logic there.